storage/src/test/java/tech/pegasys/artemis/storage/server/rocksdb/V3RocksDbDatabaseTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.storage.server.rocksdb;
+
+import static java.util.stream.Collectors.toList;
+
+import com.google.common.collect.Streams;
+import java.io.File;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import tech.pegasys.artemis.core.ChainBuilder;
+import tech.pegasys.artemis.datastructures.blocks.SignedBlockAndState;
+import tech.pegasys.artemis.datastructures.state.Checkpoint;
+import tech.pegasys.artemis.storage.Store;
+import tech.pegasys.artemis.storage.server.Database;
+import tech.pegasys.artemis.storage.server.StateStorageMode;
+
+public class V3RocksDbDatabaseTest extends AbstractRocksDbDatabaseTest {
+
+  @Override
+  protected Database createDatabase(final File tempDir, final StateStorageMode storageMode) {
+    final RocksDbConfiguration config = RocksDbConfiguration.withDataDirectory(tempDir.toPath());
+    return RocksDbDatabase.createV3(config, storageMode);
+  }
+
+  @Test
+  public void shouldHandleRestartWithUnrecoverableForkBlocks_archive(@TempDir final Path tempDir)
+      throws Exception {
+    testShouldHandleRestartWithUnrecoverableForkBlocks(tempDir, StateStorageMode.ARCHIVE);
+  }
+
+  @Test
+  public void shouldHandleRestartWithUnrecoverableForkBlocks_prune(@TempDir final Path tempDir)
+      throws Exception {
+    testShouldHandleRestartWithUnrecoverableForkBlocks(tempDir, StateStorageMode.ARCHIVE);
+  }
+
+  private void testShouldHandleRestartWithUnrecoverableForkBlocks(
```

Comment:
Doesn't need to be addressed in this PR but we should actually be removing those blocks from the hot state and rejecting any new blocks that build on them.   Not sure if we're just wasting memory at the moment or if we're considering them as part of fork choice and might try to switch to a fork that conflicts with our finalised blocks.

Reply:
I'm pretty sure we're just wasting memory right now.  We shouldn't be importing blocks or choosing head blocks that don't descend from the latest finalized block.  

I'll log a ticket to follow-up though.

Reply:
Ticket: https://pegasys1.atlassian.net/browse/BC-353